src/components/error_tooltip/error_tooltip.ts

```diff
@@ -29,28 +30,64 @@ export interface ErrorToolTipMessage {
 }
 
 interface ErrorToolTipProps {
+  cellPosition: CellPosition;
   errors: ErrorToolTipMessage[];
   onClosed?: () => void;
 }
 
-export class ErrorToolTip extends Component<ErrorToolTipProps> {
+export class ErrorToolTip extends Component<ErrorToolTipProps, SpreadsheetChildEnv> {
   static maxSize = { maxHeight: ERROR_TOOLTIP_MAX_HEIGHT };
   static template = "o-spreadsheet-ErrorToolTip";
   static props = {
+    cellPosition: Object,
     errors: Array,
     onClosed: { type: Function, optional: true },
   };
+
+  get evaluationError() {
+    const cell = this.env.model.getters.getEvaluatedCell(this.props.cellPosition);
+    if (cell.message) {
+      return cell;
+    }
+    return undefined;
+  }
+
+  get errorOriginPositionString() {
+    const evaluationError = this.evaluationError;
+    const position = evaluationError?.errorOriginPosition;
+    if (!position || deepEquals(position, this.props.cellPosition)) {
+      return "";
+    }
+    const sheetId = position.sheetId;
+    return this.env.model.getters.getRangeString(
+      this.env.model.getters.getRangeFromZone(sheetId, positionToZone(position)),
+      this.env.model.getters.getActiveSheetId()
+    );
+  }
+
+  selectCell() {
+    const position = this.evaluationError?.errorOriginPosition;
+    if (!position) {
+      return;
+    }
+    const activeSheetId = this.env.model.getters.getActiveSheetId();
+    if (position.sheetId !== activeSheetId) {
+      this.env.model.dispatch("ACTIVATE_SHEET", {
+        sheetIdFrom: activeSheetId,
+        sheetIdTo: position.sheetId,
+      });
+    }
+    this.env.model.selection.selectCell(position.col, position.row);
+  }
 }
 
 export const ErrorToolTipPopoverBuilder: PopoverBuilders = {
   onHover: (position, getters): CellPopoverComponent<typeof ErrorToolTip> => {
     const cell = getters.getEvaluatedCell(position);
     const errors: ErrorToolTipMessage[] = [];
+    let evaluationError: EvaluatedCell | undefined;
     if (cell.type === CellValueType.error && !!cell.message) {
-      errors.push({
-        title: _t("Error"),
-        message: cell.message,
-      });
+      evaluationError = cell;
     }
 
     const validationErrorMessage = getters.getInvalidDataValidationMessage(position);
```

Comment:
just inject that value directly to the errorTooltip component as you did with the EvaluationError. The builder is no longer relevant if it only manages a single case. Or you could keep the builder and have it yield a component... but the former seems easier